### PR TITLE
Autosubscribe to new log groups after stack creation

### PR DESCRIPTION
**Pull request by yanscalyr**
This requires that the account has CloudTrail logging enabled
which means having at least one trail for management events.
Having only one trail for management events is free (excluding
the S3 costs for actually storing the trail)

### CODE REVIEW

File: cloudwatch_logs/lambda/cloudwatch_subscriber/main.py [hunk 1]
```diff
@@ -372,6 +374,29 @@ def process_cf_event(event):
                 delete_subscription_filter(log_group_name, options)
                 lambda_remove_permission(log_group_name, destination_arn)
 
+def process_log_event(event):
```
Comment (czerwingithub) [inline cloudwatch_logs/lambda/cloudwatch_subscriber/main.py]: This method's name is too generic.  Let's call it `process_create_log_group_event`.

File: cloudwatch_logs/lambda/cloudwatch_subscriber/main.py [hunk 2]
```diff
@@ -397,10 +422,17 @@ def timeout_handler(sig_num, frame):
...
     except:
-        send_resp(event, context, "FAILED")
+        if send_response:
+            send_resp(event, context, "FAILED")
     else:
-        send_resp(event, context, "SUCCESS")
-        signal.alarm(0)
+        if send_response:
+            send_resp(event, context, "SUCCESS")
+            signal.alarm(0)
```
Comment (czerwingithub) [inline cloudwatch_logs/lambda/cloudwatch_subscriber/main.py]: Do you know why we signal an alarm here?  I know you are just using the existing code.

Comment (yanscalyr) [reply cloudwatch_logs/lambda/cloudwatch_subscriber/main.py]: `signal.alarm(0)` disables any previously set alarms, and we set one on line 398 to signal a timeout. No idea if this is actually necessary.

Comment (czerwingithub) [reply cloudwatch_logs/lambda/cloudwatch_subscriber/main.py]: Ah, that makes sense.  Yes, we should leave that.  That alarm is being set to guard against the request taking too long.

Comment (yanscalyr) [reply cloudwatch_logs/lambda/cloudwatch_subscriber/main.py]: Sorry line 423, was looking at the wrong branch.

**Pull request merged**